client/dashboard/me/profile-personal-details/update-email/email-verification-banner.tsx

```diff
@@ -0,0 +1,185 @@
+import { resendEmailVerificationMutation } from '@automattic/api-queries';
+import { useMutation } from '@tanstack/react-query';
+import { Link } from '@tanstack/react-router';
+import { Button } from '@wordpress/components';
+import { useDispatch } from '@wordpress/data';
+import { createInterpolateElement } from '@wordpress/element';
+import { __, sprintf } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { useState, useEffect } from 'react';
+import Notice from '../../../components/notice';
+import type { UserSettings } from '@automattic/api-core';
+
+// Get email verification params from URL
+function getEmailVerificationParams() {
+	if ( typeof window === 'undefined' ) {
+		return {
+			isEmailChangeComplete: false,
+			isEmailVerificationComplete: false,
+			emailChangeFailed: false,
+			emailVerificationFailed: false,
+		};
+	}
+	const params = new URLSearchParams( window.location.search );
+	const newEmailResult = params.get( 'new_email_result' );
+	const verified = params.get( 'verified' );
+
+	return {
+		isEmailChangeComplete: newEmailResult === '1',
+		isEmailVerificationComplete: verified === '1',
+		emailChangeFailed: newEmailResult === '0',
+		emailVerificationFailed: verified === '0',
+	};
+}
+
+// Clean up email verification params from URL
+function cleanUpEmailVerificationParams() {
+	const params = new URLSearchParams( window.location.search );
+	params.delete( 'new_email_result' );
+	params.delete( 'verified' );
+	const newUrl = window.location.pathname + ( params.toString() ? '?' + params.toString() : '' );
+	window.history.replaceState( {}, '', newUrl );
+}
+
+interface EmailVerificationBannerProps {
+	userData: UserSettings;
+}
+
+export default function EmailVerificationBanner( { userData }: EmailVerificationBannerProps ) {
+	const { createErrorNotice } = useDispatch( noticesStore );
+	const [ showResendButton, setShowResendButton ] = useState( true );
+
+	// Extract verification params once to avoid multiple URL parsing calls
+	const verificationParams = getEmailVerificationParams();
```

Comment:
```suggestion
	const {isEmailVerificationComplete,isEmailChangeComplete} = getEmailVerificationParams();
```

Something like this would probably help with the `verificationParams` repetition below (not a requirement, but I thought it was a little verbose).

Reply:
Ahh yes, gotta love destructuring! Thanks for the suggestion! Applied in [2a75a25](https://github.com/Automattic/wp-calypso/pull/106103/commits/2a75a2594fc9a0714c3590f0f16c360a85fc5f8e)!